ui/lib/utils.ts

```diff
@@ -161,3 +169,122 @@ export const convertImage = (image: string) => {
   //one slash docker images w/o docker.io
   return `https://hub.docker.com/r/${prefix}/${noTag}`;
 };
+
+// getSourceRefForAutomation returns the automation's sourceRef
+// depending on whether the automation is a Kustomization or a HelmRelease.
+export function getSourceRefForAutomation(
+  automation?: Automation
+): FluxObjectRef | undefined {
+  return automation?.kind === FluxObjectKind.KindKustomization
+    ? (automation as Kustomization)?.sourceRef
+    : (automation as HelmRelease)?.helmChart?.sourceRef;
+}
+
+// findNode searches for a node by the node's name and namespace.
```

Comment:
Is it worth just putting the nodes into a map, and then we wouldn't need to search?

Reply:
Done.